### PR TITLE
docs: added how to test createSelector functions

### PR DESCRIPTION
**Pull request by guilhermechiara**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
There are no guidelines on how to test selectors created dynamically.

Issue Number: N/A

## What is the new behavior?
Created a new small section in the docs to show how to test dynamic selectors.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by HoverBaum** [comment]
Super awesome PR, exactly the information I was just now missing in the docs.

Maybe a sentence about selectors with multiple states though.

File: docs/recipes/unit-testing.md [hunk 1]
```diff
@@ -81,3 +81,28 @@ describe('Zoo', () => {
...
+
+Usually in your application you have selectors created dynamically using the createSelector function:
+
+```TS
+export class ZooSelectors {
+  static animals = () => {
+    return createSelector([ZooState], (state: ZooStateModel) => {
+      return state.animals;
+    });
+  }
```
Comment (markwhitfeld) [inline docs/recipes/unit-testing.md]: I think that this example could be improved. You don't need to use the `createSelector` function when you are creating a selector that is not parameterised. You can just use the `@Selector` decorator. Your code here wuold better be written as:
```TS
@Selector([ZooState])
animals(state: ZooStateModel) {
  return state.animals;
}
```
So I think that a better example of a `createSelector` usage would be:
```TS
  static animalNames = (type: string) => {
    return createSelector([ZooState], (state: ZooStateModel) => {
      return state.animals
        .filter((animal) => animal.type === type )
        .map((animal => animal.name ));
    });
  }
```
The test would need to be adjusted accordingly

File: docs/recipes/unit-testing.md [hunk 2]
```diff
@@ -81,3 +81,28 @@ describe('Zoo', () => {
...
+
+```TS
+it('should select animals from state', () => {
+  const value = ZooSelectors.animals()({
+    animals: ['zebra', 'panda']
+  });
+    
+  expect(value).toEqual(['zebra', 'panda']);
+});
+```
```
Comment (markwhitfeld) [inline docs/recipes/unit-testing.md]:
```TS
it('should select animals from state', () => {
  const value = ZooSelectors.animalNames('zebra')({
    animals: [ 
      { type: 'zebra', name: 'Andy'},
      { type: 'panda', name: 'Betty'},
      { type: 'zebra', name: 'Crystal'},
      { type: 'panda', name: 'Donny'},
    ]
  });
    
  expect(value).toEqual(['Andy', 'Crystal']);
});
```

**Comment by markwhitfeld** [comment]
Ah man! I see the @splincode merged the PR while I was doing my review...
I'll make the changes ;-)

**Comment by splincode** [comment]
@markwhitfeld Sorry :)

**Comment by markwhitfeld** [comment]
No worries. Github should really have something that warns if there is a review in progress when you initiate a merge.
PS I see that the commit message was in past tense, it should be present tense. But to be honest I often don't notice that either.

**Pull request merged**